libs/labelbox/src/labelbox/schema/user_group.py

```diff
@@ -0,0 +1,523 @@
+        """
+        return self._projects
+
+    @projects.setter
+    def projects(self, value: Set[Union[UserGroupProject, Project]]) -> None:
+        """
+        Sets the list of project IDs in the group.
+
+        Args:
+            value (Set[GroupProject]): The list of project IDs to set.
+        """
+        self._projects = value
+
+    def update(self) -> "UserGroup":
+        """
+        Updates the group in Labelbox.
+
+        Returns:
+            UserGroup: The updated UserGroup object. (self)
+
+        Raises:
+            UnprocessableEntityError: If the update fails.
+        """
+        query = """
+        mutation UpdateUserGroupPyApi($id: ID!, $name: String!, $color: String!, $projectIds: [String!]!, $userIds: [String!]!) {
+            updateUserGroup(
+                where: {id: $id}
+                data: {name: $name, color: $color, projectIds: $projectIds, userIds: $userIds}
+            ) {
+                group {
+                    id
+                    name
+                    color
+                    projects {
+                        nodes {
+                            id
+                            name
+                        }
+                    }
+                    members {
+                        nodes {
+                            id
+                            email
+                        }
+                    }
+                }
+            }
+        }
+        """
+        params = {
+            "id":
+                self.id,
+            "name":
+                self.name,
+            "color":
+                self.color.value,
+            "projectIds": [
+                project.id if hasattr(project, 'id') else project.uid
+                for project in self.projects
+            ],
+            "userIds": [
+                user.id if hasattr(user, 'id') else user.uid
+                for user in self.users
+            ]
+        }
+        result = self.client.execute(query, params)
+        if not result:
+            raise UnprocessableEntityError("Failed to update group")
+        return self
+
+    def create(self) -> "UserGroup":
+        """
+        Creates a new group in Labelbox.
+
+        Args:
+            client (Client): The Labelbox client.
+            name (str): The name of the group.
+            color (GroupColor, optional): The color of the group. Defaults to GroupColor.BLUE.
+            users (List[User], optional): The users to add to the group. Defaults to [].
+            projects (List[Project], optional): The projects to add to the group. Defaults to [].
+
+        Returns:
+            Group: The newly created group. (self)
+
+        Raises:
+            ResourceCreationError: If the group already exists or if the creation fails.
+            ValueError: If the group name is not provided.
+        """
+        if self.id:
+            raise ResourceCreationError("Group already exists")
```

Comment:
seems like this error should come from the server?

Reply:
I disagree

Example 
UserGroup(id="1", name="mygroup").create()  = create { name="mygroup", id="3"} 
UserGroup(name="mygroup")  = create {name ="mygroup", id="2"}

The only thing unique to a group it's an id, two groups can have the same name.

The createApiEndpoint would need to have an id exposed to validate this or you'd have to fetch twice (once to check and then once to create if the fetch results in no group found). Upsert behavior, to avoid this two call tax, which doesn't exist and would like to avoid creating.

Simple is 0 calls. If you provide an id on usergroup initialization, the state is fetched and creating an existing usergroup again makes no sense.